### ISSUE TITLE
[C#] Accented characters in C# comments cause Unicode UTF-8 parsing errors

### ISSUE THREAD

**Issue by tetrapod00** [godotengine/godot#96703]
### Tested versions

Reporting this on behalf of someone else.
**Bug reproduces on their specs:**
Windows 11
.NET version 8.0.303
VSCode as text editor
Godot Engine v4.3.stable.mono.official.77dcf97d8 - https://godotengine.org
Vulkan 1.3.260 - Forward+ - Using Device #0: AMD - AMD Radeon(TM) Graphics

**Does not reproduce on my specs:**
Windows 10
.NET version 8.0.303
VSCode as text editor
Godot v4.3.stable.mono - Windows 10.0.19045 - GLES3 (Compatibility) - NVIDIA GeForce GTX 1660 Ti with Max-Q Design (NVIDIA; 32.0.15.5612) - AMD Ryzen 7 3750H with Radeon Vega Mobile Gfx (8 Threads)

### System information

See above

### Issue description

I'm reporting this on behalf of someone who does not have access to github. When using accented characters in a C# script, the script will not compile, and produces UTF-8 errors. This is the script, placed on a single Node2D. I did suspect `Node2d` as a class name, but it seems to not affect things.
```
using Godot;
using System;

public partial class Node2d : Node2D
{
    // Called when the node enters the scene tree for the first time.
    public override void _Ready() { }

    // Called every frame. 'delta' is the elapsed time since the previous frame.
    public override void _Process(double delta)
    {
        // Apenas um único comentário para verificar a detecção de erros
        GD.Print("Hello world");
    }
}
```
The comment which causes the problem: `// Apenas um único comentário para verificar a detecção de erros`

The error does not reproduce when using accented characters with GDScript.

Console log:
![console_log](https://github.com/user-attachments/assets/64e5d188-e685-4388-8799-a38b953b86b5)
Debugger log:
![deugger_log](https://github.com/user-attachments/assets/1d7d84e3-cfe6-4b7b-8132-3993f7573132)

Unfortunately, I can't reproduce the same result. My console log when running this MRP:
```
Godot Engine v4.3.stable.mono.official (c) 2007-present Juan Linietsky, Ariel Manzur & Godot Contributors.
--- Debugging process started ---
Godot Engine v4.3.stable.mono.official.77dcf97d8 - https://godotengine.org
OpenGL API 3.3.0 NVIDIA 556.12 - Compatibility - Using Device: NVIDIA - NVIDIA GeForce GTX 1660 Ti with Max-Q Design

Hello world
Hello world
Hello world
...
```
The differences between my specs and theirs are `Windows 10` vs `Windows 11`, and potentially VSCode configuration, Windows configuration, or .NET configuration. Maybe some settings in one of these places is affecting how accented characters are saved in text or .cs files?

### Steps to reproduce

Open the MRP and run the game.

### Minimal reproduction project (MRP)

[mrp-csharp-comment.zip](https://github.com/user-attachments/files/16920906/mrp-csharp-comment.zip)

**Comment by hawkerm** [comment]
I was able to build with the comment copied in a C# script of mine, from VS 2022 17.11.1 on my Win 10 19045.4780 machine with the comment as well .NET 8.0.400.

However, if I built/ran from Godot (it seemed to run), but I saw this in the output:

```
  Unicode parsing error, some characters were replaced with � (U+FFFD): Invalid UTF-8 continuation byte (fa ... 6e ...)
  Unicode parsing error, some characters were replaced with � (U+FFFD): Invalid UTF-8 continuation byte (e1 ... 72 ...)
  Unicode parsing error, some characters were replaced with � (U+FFFD): Invalid UTF-8 continuation byte (e7 ... e3 ...)
Godot Engine v4.3.stable.mono.official.77dcf97d8 - https://godotengine.org
Vulkan 1.3.280 - Forward Mobile - Using Device #0: NVIDIA - NVIDIA GeForce GTX 1080

  Unicode parsing error, some characters were replaced with � (U+FFFD): Invalid UTF-8 continuation byte (fa ... 6e ...)
  Unicode parsing error, some characters were replaced with � (U+FFFD): Invalid UTF-8 continuation byte (e1 ... 72 ...)
  Unicode parsing error, some characters were replaced with � (U+FFFD): Invalid UTF-8 continuation byte (e7 ... e3 ...)
```

![image](https://github.com/user-attachments/assets/6ec5b327-a0e6-40ed-825a-c73bd124ed93)

![image](https://github.com/user-attachments/assets/e5905f07-e83c-463c-bd99-5871dab2f61a)

**Comment by rboram** [comment]
Hi, I'm the person who "discovered" this bug. If you need any help or questions, I'm here to help!

**Comment by rboram** [comment]
Hi, I'm the person mentioned who reported the bug on discord. If you need help trying to replicate the error or anything else, I'm available! Thank you @tetrapod00

**Comment by bruvzg** [comment]
Not reproducible with MRP, it contains valid UTF-8 script. But it is (and invalid characters are the same) if you resave it as Windows-1252 encoding. Godot only supports UTF-8 make sure to save files in it.

In case of VSCode, encoding is can be changed in the right bottom corner.

<img width="439" alt="Screenshot 2024-09-08 at 16 35 05" src="https://github.com/user-attachments/assets/d1ef5327-734a-42d0-96e5-408ad5c4c879">

**Comment by tetrapod00** [comment]
Yep, I can reproduce it with the MRP if the file is re-saved with `Windows 1252`.

**Comment by bruvzg** [comment]
The underlying issue is likely mostly in the VSCode config, but it might be also related to the fact Godot do not use byte order mark for UTF-8 (Godot can read files with BOM, but is not adding to the saved files). Using BOM for UTF-8 is not required and not recommended by Unicode, but it might help other apps to correctly identify files created by Godot as UTF-8.

**Comment by hawkerm** [comment]
As mentioned, I also reproduced by modifying the file in Visual Studio 2022 (not VSCode) as well (though not sure where it's file encoding option is, it's not in the statusbar like VSCode). Though if I modified the file within Godot itself without an external editor set, then it seemed fine.

Can confirm in VSCode though saving as UTF-8 resolves the issue, compared to Windows 1252.

Seems like the error message about ensuring scripts are saved in UTF8 is helpful here, but it should it link to a doc page or something that helps folks to understand how to do that in various editors like VS Code, VS 2022, Rider, etc...?
